pyanaconda/kickstart.py

```diff
@@ -80,12 +80,32 @@
 from pykickstart.version import returnClassForVersion
 
 import logging
-log = logging.getLogger("anaconda")
+anaconda_log = logging.getLogger("anaconda")
+script_log = logging.getLogger("anaconda.kickstart.script")
+parsing_log = logging.getLogger("anaconda.kickstart.parsing")
 stderrLog = logging.getLogger("anaconda.stderr")
 storage_log = logging.getLogger("blivet")
 stdoutLog = logging.getLogger("anaconda.stdout")
 from pyanaconda import anaconda_log
 
+# command specififc loggers
+authconfig_log = logging.getLogger("anaconda.kickstart.authconfig")
```

Comment:
That sounds good - and it indeed looks like our kickstart command subclasses look like a good use case for the class instance log (eq. just call self.log.info() and the appropriate prefix will be added automatically based on the class). But as we subclass the kickstart commands in this case, it will probably have to a mixin that adds the logging functionality to the given kickstart command class.

BTW, a potential show-stopper - looks like the instantiated kickstart command classes for some reason get serialized when a storage snapshot is being created and I've seen it die when it tried to serialize a logger instance (eq. self.log = getLogger("kickstart.timezone") was enough to make it crash during serialization). It might not be an issue if self.log is is a proper property, or the serialization could be fixed to skip the kickstart command instances, but it's certainly something to watch for.

For that reason I would like to be careful and merge this method first (statically defined loggers on top of file) before investigating using the class instance loggers as there is less chance of immediate breakage like this.